part2/test_diagonalization.ipynb

```
@@ -0,0 +1,231 @@
+        "\n",
+        "    # Case 1: 2×2 tam giác trên tổng quát\n",
+        "    A1 = [[4, 1], [0, 2]]\n",
+        "    e1, _, _, _ = reconstruction_rel_err(A1)\n",
+        "    print(f\"1. Upper-triangular 2x2: rel_err={e1:.3e} (expected < 1e-10)\")\n",
+        "\n",
+        "    # Case 2: Đối xứng\n",
+        "    A2 = [[2, 1], [1, 2]]\n",
+        "    e2, _, _, _ = reconstruction_rel_err(A2)\n",
+        "    print(f\"2. Symmetric: rel_err={e2:.3e} (expected < 1e-10)\")\n",
+        "\n",
+        "    # Case 3: Đường chéo — trị riêng 5 và 3\n",
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)